docs/Hooks.md

```diff
@@ -122,19 +121,38 @@ fastify.addHook('onSend', (request, reply, payload, next) => {
 })
 ```
 
+## Application Hooks
+
+You are able to hook into the application-lifecycle as well. It's important to note that these hooks aren't fully encapsulated. The `this` inside the hooks are encapsulated but the handlers can respond to an event outside the encapsulation boundaries.
+
+- `'onClose'`
+- `'onRoute'`
+
 <a name="on-close"></a>
 **'onClose'**<br>
-The unique hook that is not inside the lifecycle is `'onClose'`, this one is triggered when you call `fastify.close()` to stop the server, and it is useful if you have some [plugins](https://github.com/fastify/fastify/blob/master/docs/Plugins.md) that need a "shutdown" part, such as a connection to a database.<br>
-Only for this hook, the parameters of the function changes, the first one is the Fastify instance, the second one the `done` callback.
+This one is triggered when you call `fastify.close()` to stop the server, and it is useful if you have some [plugins](https://github.com/fastify/fastify/blob/master/docs/Plugins.md) that need a "shutdown" part, such as a connection to a database.<br>
+The first argument is the Fastify instance, the second one the `done` callback.
 ```js
 fastify.addHook('onClose', (instance, done) => {
   // some code
   done()
 })
 ```
+<a name="on-route"></a>
+**'onRoute'**<br>
+This one is called when your register a new route.<br>
+The first argument are the `route` options. The interface is synchronous and you don't have to call a callback.
+```js
+fastify.addHook('onRoute', (route) => {
```

Comment:
IMO `route` should be named `options` so that it's consistent with the `fastify.route(options)` definition from the [Routes documentation](https://github.com/fastify/fastify/blob/master/docs/Routes.md#full-declaration). This will help people understand that it's the same object.

Reply:
It depends on the context it is also documented `The first argument are the `route` options ...`

Reply:
It's easier to understand code than regular language sometimes :)
How about `routeOptions`?

Reply:
Actually that's a little long. I think it's fine.